src/opm/output/eclipse/AggregateGroupData.cpp

```diff
@@ -627,6 +615,24 @@ void injectionGroup(const Opm::Schedule&     sched,
     }
 }
 
+template <class IGrpArray>
+void storeNodeSequenceNo(const Opm::Schedule& sched,
+                    const Opm::Group& group,
+                    const int nwgmax,
+                    const std::size_t simStep,
+                    IGrpArray& iGrp) {
+
+    using IGroup = ::Opm::RestartIO::Helpers::VectorItems::IGroup::index;
+
+    const auto& netwrk = sched[simStep].network();
+    const auto seq_ind = findInVector(netwrk.node_names(), group.name());
+
+    // The igrp node number is equal to the node sequence number from the BRANPROP keyword
+    // for the groups that also are nodes in the external network (BRANPROP, NODEPROP)
+    // If not - the node numbr is zero.
+    iGrp[nwgmax + IGroup::NodeNumber] = seq_ind ? seq_ind.value()+1 : 0;
```

Comment:
Here you could use the `std::optional<T>::value(const T& default)` overload:
```
iGrp[] = seq_ind.value( -1 ) + 1;
```

Reply:
I suggest to keep the original code because your suggestion seems not to be the common / standard use of std::optional. I have difficulties finding proper documentation of your suggested use on the internet.

Reply:
Ehhh - you are absolutely right - I meant the `value_or( ..)` method.